File: Java/Problem-Solving/powerOf2.java
//find if no is power of 2 or not
import java.util.Scanner;

public class powerOf2 {

	
		   public static boolean isPowerOf2(int n)
		    {
		        if (n == 0)
		            return false;
		        while(n != 1)
                {
                    if(n%2 != 0)
                        return false;
                    n = n/2;
                }
               return true;
		        
		    }

		
		    public static void main(String[] args)
		    {
		    	Scanner uinput = new Scanner(System.in);
                int limit = uinput.nextInt();
                int[] nValues = new int[limit];
                for(int i=0; i<limit; i++)
                {
                    nValues[i] = uinput.nextInt();
                }
                
                for(int i=0; i<limit; i++)
                {
                    if(isPowerOf2(nValues[i]))
                       System.out.println("YES");
                    else
                       System.out.println("NO");
                       
                }
		       

		        


		    }
}

File: Python/Problem-Solving/PowerOf2.py
#Find if number is power of 2
# Qus asked in Oracle, Intel, apnabazar, Microsoft
#difficulty : Basic
#py3 Code

#this function checks by dividing no by 2 if n becomes 1 then it is power of 2 otherwise not a power of 2
def isPowerOf2(n):
    if n == 0:
        return False
    while n != 1:
        if n % 2 != 0:
            return False
        n = n//2
    return True

nValues = []
limit = int(input())
#Taking nValues
for i in range(limit):
     nValues.append(int(input()))
#checking if isPowerOf2 function returns true or false if returns true prints yes otherwise no
for i in nValues:
    if (isPowerOf2(i)):
        print("YES")
    else:    
        print("NO")  